src/main/java/seedu/club/logic/commands/ChangeTaskStatusCommand.java

```diff
@@ -0,0 +1,111 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.lang3.text.WordUtils;
+
+import seedu.club.commons.core.Messages;
+import seedu.club.commons.core.index.Index;
+import seedu.club.logic.commands.exceptions.CommandException;
+import seedu.club.logic.commands.exceptions.IllegalExecutionException;
+import seedu.club.model.task.Assignee;
+import seedu.club.model.task.Assignor;
+import seedu.club.model.task.Date;
+import seedu.club.model.task.Description;
+import seedu.club.model.task.Status;
+import seedu.club.model.task.Task;
+import seedu.club.model.task.Time;
+import seedu.club.model.task.exceptions.DuplicateTaskException;
+import seedu.club.model.task.exceptions.TaskNotFoundException;
+
+/**
+ * Edits the status of a existing task in the club book.
+ */
+public class ChangeTaskStatusCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "changestatus";
+    public static final ArrayList<String> COMMAND_ALIASES = new ArrayList<>(
+            Arrays.asList(COMMAND_WORD, "status")
+    );
+    public static final String COMMAND_FORMAT = COMMAND_WORD + " INDEX s/STATUS";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Modifies the status of the task identified "
+            + "by the index number used in the last task listing. "
+            + "The existing status will be overwritten by the input status.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "st/STATUS\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_STATUS + "Completed";
+
+    public static final String MESSAGE_INVALID_PERMISSION = "This task's status cannot be updated "
+            + "as you are not the assignee!";
+    public static final String MESSAGE_CHANGE_SUCCESS = "Status of task - %1$s successfully changed!";
+    public static final String MESSAGE_NOT_CHANGED = "Status of task unchanged!";
+
+    private final Index index;
+    private Task taskToEdit;
+    private Task editedTask;
+    private final Status newStatus;
+
+    public ChangeTaskStatusCommand(Index index, Status newStatus) {
+        this.index = index;
+        String status = newStatus.getStatus();
+        String capitalizedStatus = WordUtils.capitalize(status);
+        this.newStatus = new Status(capitalizedStatus);
+
+    }
+
+    @Override
+    protected void preprocessUndoableCommand() throws CommandException {
+        List<Task> lastShownList = model.getFilteredTaskList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        taskToEdit = lastShownList.get(index.getZeroBased());
+        editedTask = createEditedTask(taskToEdit);
+    }
+
+    @Override
+    protected CommandResult executeUndoableCommand() throws CommandException {
+        try {
+            model.changeStatus(taskToEdit, editedTask);
```

Comment:
add support for requireToSignUp and requireToLogIn depending on @th14thmusician 's implementation

Reply:
will do this once @th14thmusician 's PR is merged.